### ISSUE TITLE
Train Problem

### ISSUE THREAD

**Issue by qichengxxxx** [zjzser/TraceableSpeech#2]
![Image](https://github.com/user-attachments/assets/f37649bb-2377-4060-9afd-d1e39f2b664c)
Hello, I'm training this model with a 40G A100GPU, and the dataset only uses train-clean-100. The batch_size is 32 and the num_workers are 6. Why is my training so slow? Every time I train at 500steps, it gets stuck for a long time. I know it's the verification process. Does the verification process take that long?

**Comment by qichengxxxx** [comment]
By the way, why does an error always occur after verification?

**Comment by zjzser** [comment]
that is a privacy error. From this pic, it seems that except this, all is fine. I suggest that you can control the valiation set small to debug it. Well, but I would say, every updata step s/b is 0.922, 0.968 and so on, that is normal. as for a epoch time 10201 second（2.83 h）it is normal too. in my experience, if you mel-loss low to maybe about 0.6, the watermark is can nearly successful ,and that will spend me at least 1.5 days. as for the audio error, i suggest find the reason from the front-end preprocessing.

**Comment by qichengxxxx** [comment]
Hi, I have only been exposed to the voice field recently. Is this research only aimed at the Codec module? There is no TTS-related content in the project, right?

**Comment by zjzser** [comment]
well, yes, after you trained this codec, you can use in different LM model of TTS

**Comment by qichengxxxx** [comment]
![Image](https://github.com/user-attachments/assets/645a5ed3-08c6-4952-b52f-92464ad53894)
OK, like your paper, does it refer to these two modules?

**Comment by qichengxxxx** [comment]
I really want to understand how you combined it, and I don't know if I can provide code based on the Vall-E model, thank you very much. This is my email address: ruier0041@gmail.com

**Comment by zjzser** [comment]
well, if you training the codec, you just follow the valle-pipeline and transfer the codec(usually like encodec) to this one, The valle code that i used is https://github.com/lifeiteng/vall-e.

**Comment by qichengxxxx** [comment]
Is it correct to replace like the following? I am confused about how to configure the model parameters after replacement.
After replacement: Original audio → [encoder] → token sequence → [decoder] → Synthetic audio.

**Comment by zjzser** [comment]
yes，you are right

**Comment by qichengxxxx** [comment]
![Image](https://github.com/user-attachments/assets/df6763ce-f923-4cf6-b538-85db2dc1f940)|

I trained with the train-other-500 dataset, and verified it with the dev-clean dataset, and finally tested it with test-clean. The test results are as follows. Is this result in expectations?

**Comment by moral66** [comment]
> ![Image](https://github.com/user-attachments/assets/df6763ce-f923-4cf6-b538-85db2dc1f940)|
> 
> I trained with the train-other-500 dataset, and verified it with the dev-clean dataset, and finally tested it with test-clean. The test results are as follows. Is this result in expectations?

What is the SNR you tested? I used the same training set as you, but my 4090 GPU with 24g memory is not enough to use batch=32. At 150000 steps, the ACC was only 50+. When I trained to 480000, the ACC index was almost the same as yours. I am studying how to shorten the training time.

**Comment by zjzser** [comment]
The result is match the expectations. 
The paper recorded the SNR and other value. 
You should train enough step, and you can check the tensorboard to follow the predict result.

**Comment by qichengxxxx** [comment]
@moral66 I modified the batch size of the verification process, thereby shortening the training time.
